SimonImanigirimpuhwe/my-brand-website-api
# #11 ft(article): enable post article with image

#### What does this PR do?

- Enable post  article  with image

#### Description of Task to be completed? 

- import multer to the post endpoint

- Add articleImage schema property to the article schema

#### How should this be manually tested?

- Clone the repo 

- In your home directory  cd this branch `ft-upload-article-image-#10`

- Then run `npm install` to install all dependencies

- Run `npm run dev` to start the server

- Use postman send a `POST` request on `http://localhost:3000/articles` 

> In the request body send `{
        title :article title,
       content: article content,
       articleImage: select image file
}`

> You should see a returned result with success message and with  the posted article with image


#### Any background context you want to provide?

- To be connected to DB you need to check `env-sample` for the credentials settings

- To be able to update an article you  need to **login**

- To post article with image you need to choose in the body  `form-data` and select file along with the article content and title

- Multer needs to be installed

#### What are the relevant Trello stories?

- [#11](https://trello.com/c/mPSSbtcf/11-enable-post-article-with-image)

#### Screenshots (if appropriate)

<img width="1440" alt="post-article-img" src="https://user-images.githubusercontent.com/54619678/96671282-b9821e80-1361-11eb-8a12-748730e67b6a.png">


#### Questions:

- N/A
